competition/track2/train/train.py

```diff
@@ -87,36 +87,30 @@ def train(input_path, output_path):
         for id in vehicle_ids[0:n_vehicles]:
             print(f"Adding data for vehicle id {id} in scenario {scenario}.")
 
-            with open(
-                scenario_path / (f"Agent-history-vehicle-{id}.pkl"),
-                "rb",
-            ) as f:
-                vehicle_data = pickle.load(f)
+            for pkl_file in os.listdir(scenario_path):
+                if pkl_file.endswith(f"-{id}.pkl"):
+                    with open(scenario_path / pkl_file, "rb") as f:
+                        vehicle_data = pickle.load(f)
             image_names = list()
-
             for filename in os.listdir(scenario_path):
                 if filename.endswith(f"-{id}.png"):
                     image_names.append(filename)
 
             image_names = sorted(image_names)
-
-            goal_pos_x = vehicle_data[float(image_names[-1].split("_Agent")[0])]["ego"][
-                "pos"
-            ][0]
-            goal_pos_y = vehicle_data[float(image_names[-1].split("_Agent")[0])]["ego"][
-                "pos"
-            ][1]
+            
+            goal_pos_x = vehicle_data[float(image_names[-1].split("_Agent")[0])].ego_vehicle_state.mission.goal.position.x
+            goal_pos_y = vehicle_data[float(image_names[-1].split("_Agent")[0])].ego_vehicle_state.mission.goal.position.y
```

Comment:
@cz0204 @alfredgu001324 
1) Try to avoid using "Agent" to get the timestamp.
2) Try to only use the file naming format, shown below, as the only information available in order to extract any info.
    ![image](https://user-images.githubusercontent.com/24203326/189713402-252e4d17-38cd-4996-8d16-2395369120f3.png)
3) With respect to
    ```python
    vehicle_data[float(image_names[-1].split("_Agent")[0])].ego_vehicle_state.mission.goal.position.x
    ```
    + Why did we use the data from the last time point to get the `mission.goal.position` value?
    + I believe that the `mission.goal.position` value is the same at all time points for a given vehicle.
    + Could you check on this matter and simplify the code if necessary?

Reply:
1. fixed
2. fixed
3. The mission.goal.position is indeed the same at all times, so picking any arbitrary time would be fine. I changed it to the initial starting time but it doesn't really matter that much.